k8sutils/agesecret.go

```diff
@@ -137,7 +144,11 @@ func CheckAgeKeyReference(ageSecret *v1alpha1.AgeSecret, k8sclient client.Client
 }
 
 func cloneLabels(labels map[string]string) map[string]string {
-	return cloneMap(labels)
+	tmpLabels := cloneMap(labels)
```

Comment:
```go
newMap := make(map[string]string)

for key, value := range oldMap {
	// if key is not in unwanted labels add it to newMap
}

return newMap
```

Reply:
I think this is an optimization to consider, but reduces code readability, as this function does one thing and is shared between "cloneLabels" and "cloneAnnotations", combining the logic of filtering with this function may add complexity

Reply:
I meant changing the `cloneLabels` function, not the `cloneMap` function. That is, lines 147 to 151.